### PR TITLE
CODEOWNERS: cleanup and remove duplicates

### PR DESCRIPTION
**Pull request by nashif**
Remove anything that would be covered by MAINTAINERS.yml leaving only
mostly instances that are not covered.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

### CODE REVIEW

File: CODEOWNERS [hunk 1]
```diff
@@ -168,32 +115,20 @@
...
 /boards/arm/ubx_bmd345eval_nrf52840/      @Navin-Sankar @brec-u-blox
 /boards/arm/nrf5340_audio_dk_nrf5340      @koffes @alexsven @erikrobstad @rick1082 @gWacey
-/boards/common/                           @mbolivar-ampere
-/boards/deprecated.cmake                  @tejlmand
-/boards/mips/                             @frantony
-/boards/nios2/                            @nashif
-/boards/nios2/altera_max10/               @nashif
 /boards/arm/stm32_min_dev/                @sidcha
-/boards/posix/                            @aescolar @daor-oti
 /boards/posix/nrf52_bsim/                 @aescolar @wopu-ot
```
Comment (aescolar) [inline CODEOWNERS:119]: also covered by maintainers

**Review by aescolar — DISMISSED** [review]
All lines that relate to my areas are good

**Review by jhedberg — DISMISSED** [review]

**Review by henrikbrixandersen — DISMISSED** [review]
Looks good. Perhaps we should add a note to the top of this file pointing people towards the maintainers file instead?

**Review by jfischer-no — DISMISSED** [review]

**Review by fabiobaltieri — DISMISSED** [review]
I think that few non-maintainer contributors use the file to get cc'd on changes in specific drivers they care about. Should we support that use case or clarify that it's not really supported and we should only use the MAINTAINERS file?

File: doc/contribute/guidelines.rst [hunk 2]
```diff
@@ -651,8 +651,8 @@ workflow here:
    request for the ``main`` branch. The title and message from your commit
    message should appear as well.
 
-#. GitHub will assign one or more suggested reviewers (based on the
-   CODEOWNERS file in the repo). If you are a project member, you can
+#. A bot will assign one or more suggested reviewers (based on the
+   MAINTAINETRS file in the repo). If you are a project member, you can
```
Comment (fabiobaltieri) [inline doc/contribute/guidelines.rst:655]: typo (MAINTAINERS)

**Comment by aescolar** [comment]
> I think that few non-maintainer contributors use the file to get cc'd on changes in specific drivers they care about.

In the original MAINTAINERS file proposal ( https://github.com/zephyrproject-rtos/zephyr/pull/20591 ) we had an "inform" entry to cover this use case. I'd say we should just bring it back.

**Comment by fabiobaltieri** [comment]
> In the original MAINTAINERS file proposal ( #20591 ) we had an "inform" entry to cover this use case. I'd say we should just bring it back.

I don't think it would cover that specific use case, my impression is that those users would want to get looped in for some specific drivers they care about, but not an entire area of subsystem. Honestly I can't really think of a clean way of doing this, having such use case covered by changing a file in the repository is going to be a big mess (as the codeowner file is right now), we may as well not support that use case. Gerrit had the concept of "Watched Projects" that I think covers this use case without messing with the repository itself but I doubt GitHub supports such a thing.

**Review by fabiobaltieri — APPROVED** [review]

**Review by jfischer-no — APPROVED** [review]

**Review by aescolar — APPROVED** [review]

**Review by henrikbrixandersen — APPROVED** [review]

**Comment by nashif** [comment]
> I suggest sending a mail to zephyr-devel about the change in CODEOWNERS.

sure.

**Pull request merged**